### PR TITLE
LPS-194350 Make Sharepoint configuration instance level

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/browse/LPS-194350

Move sharepoint configuration from System to Instance level. No
upgrades are necessary, as system level configuration is still usable.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
That is the expected behaviour. Configurations are scoped at instance level, not site level.

**Comment by marcogalluzzi** [comment]
> That is the expected behaviour. Configurations are scoped at instance level, not site level.

Sorry, I wanted to say... he Sharepoint OAuth configuration created in one INSTANCE is available in the other INSTANCE... that was my test.

**Comment by adolfopa** [comment]
@marcogalluzzi Are you creating those new configurations in System Settings or instance settings? Those created in System Settings are globally available (all instances); the Instance Settings ones should be private to their instance.

**Comment by marcogalluzzi** [comment]
> @marcogalluzzi Are you creating those new configurations in System Settings or instance settings? Those created in System Settings are globally available (all instances); the Instance Settings ones should be private to their instance.

Hi @adolfopa I can confirm you that the Sharepoint configuration created at Instance Settings in one instance is available for selection when creating a Repository at D&M in a different instance.

**Comment by adolfopa** [comment]
@marcogalluzzi It should work as expected now; this required API changes, but there's no other way around it.

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]
Fine, now it works as expected!

Just some minor optional changes

File: portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalog.java [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
 	public Iterable<RepositoryClassDefinition>
-		getExternalRepositoryClassDefinitions();
+		getExternalRepositoryClassDefinitions(long companyId);
 
-	public Collection<String> getExternalRepositoryClassNames();
+	public Collection<String> getExternalRepositoryClassNames(long companyId);
 
 	public RepositoryClassDefinition getRepositoryClassDefinition(
-		String className);
+		long companyId, String className);
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalog.java:21]: In public APIs, the company ID usually goes first (there are exceptions though). I'd leave it like this for consistency with other repository interfaces.

File: portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImpl.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public String getRegistryName() {
 
 	@Override
 	public RepositoryClassDefinition getRepositoryClassDefinition(
-		String className) {
+		long companyId, String className) {
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImpl.java:92]: See my previous comment.

File: portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImpl.java [hunk 3]
```diff
@@ -64,28 +66,37 @@ public void destroy() {
 	public Iterable<RepositoryClassDefinition>
 		getExternalRepositoryClassDefinitions(long companyId) {
 
+		Collection<RepositoryClassDefinition>
+			systemExternalRepositoryClassDefinitions =
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImpl.java]: Done.

File: portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImpl.java [hunk 4]
```diff
@@ -64,28 +66,37 @@ public void destroy() {
...
 		}
 
-		return companyRepositoryClassDefinitions.values();
+		return systemExternalRepositoryClassDefinitions;
 	}
 
 	@Override
 	public Collection<String> getExternalRepositoryClassNames(long companyId) {
+		Collection<String> systemExternalRepositoryClassNames =
+			_getSystemExternalRepositoryData(Map::keySet);
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImpl.java:69]: Done.

**Comment by adolfopa** [comment]
Rebased and added an additional commit with suggested renames.

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:forward

**Comment by marcogalluzzi** [comment]
Hi @adolfopa the test class `RepositoryClassDefinitionCatalogImplTest` need to be fixed.

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:forward:force

**Comment by marcogalluzzi** [comment]
@adolfopa there is an SF issue with RepositoryClassDefinitionCatalogImplTest.java

```
     [java] java.lang.Exception: Found 1 formatting issues:
     [java] 1: /opt/dev/projects/github/liferay-portal/portal-impl/test/unit/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImplTest.java expected:<...feray.portal.kernel.[language.LanguageUtil;
     [java] import com.liferay.portal.kernel.module.util.SystemBundleUtil;
     [java] import com.liferay.portal.kernel.repository.RepositoryConfigurationBuilder;
     [java] import com.liferay.portal.kernel.repository.registry.RepositoryDefiner;
     [java] import com.liferay.portal.kernel.test.util.TestPropsValues;
     [java] import com.liferay.portal.kernel.util.ProxyUtil;
     [java] import com.liferay.portal.language.LanguageImpl;
     [java] import com.liferay.portal.test.rule.LiferayUnitTestRule;
     [java] 
     [java] import java.util.Collection;
     [java] import java.util.Objects;
     [java] 
     [java] import org.junit.After;
     [java] import org.junit.AfterClass;
     [java] import org.junit.Assert;
     [java] import org.junit.Before;
     [java] import org.junit.BeforeClass;
     [java] import org.junit.ClassRule;
     [java] import org.junit.Rule;
     [java] import org.junit.Test;
     [java] 
     [java] import org.osgi.framework.BundleContext;
     [java] import org.osgi.framework.ServiceRegistration;
     [java] 
     [java] /**
     [java]  * @author Leon Chi
     [java]  */
     [java] public class RepositoryClassDefinitionCatalogImplTest \{
     [java] 
     [java]   @ClassRule
     [java]   @Rule
     [java]   public static final LiferayUnitTestRule liferayUnitTestRule =
     [java]     LiferayUnitTestRule.INSTANCE;
     [java] 
     [java]   @BeforeClass
     [java]   public static void setUpClass() \{
     [java]     LanguageUtil languageUtil = new LanguageUtil();
     [java] 
     [java]     languageUtil.setLanguage(new LanguageImpl());
     [java] 
     [java]     _repositoryClassDefinitionCatalogImpl =
     [java]       new RepositoryClassDefinitionCatalogImpl();
     [java] 
     [java]     _repositoryClassDefinitionCatalogImpl.afterPropertiesSet();
     [java]   \}
     [java] 
     [java]   @AfterClass
     [java]   public static void tearDownClass() \{
     [java]     _repositoryClassDefinitionCatalogImpl.destroy();
     [java]   \}
     [java] 
     [java]   @Before
     [java]   public void setUp() \{
     [java]     BundleContext bundleContext = SystemBundleUtil.getBundleContext();
     [java] 
     [java]     _serviceRegistration = bundleContext.registerService(
```

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
Hi @adolfopa Could you help to confirm if the below failures are related to this PR? Thanks.

testGetExternalRepositoryClassDefinitions(com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest)
testGetRepositoryClassDefinition(com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest):
testGetExternalRepositoryClassNames(com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest)
```
 [junit] Testcase: testGetExternalRepositoryClassDefinitions(com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest):	Caused an ERROR
    [junit] null
    [junit] java.lang.NullPointerException
    [junit] 	at com.liferay.portal.kernel.service.CompanyLocalServiceUtil.getCompanyByWebId(CompanyLocalServiceUtil.java:441)
    [junit] 	at com.liferay.portal.kernel.test.util.TestPropsValues.getCompanyId(TestPropsValues.java:78)
    [junit] 	at com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest.testGetExternalRepositoryClassDefinitions(RepositoryClassDefinitionCatalogImplTest.java:85)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    [junit] 
    [junit] 
    [junit] Testcase: testGetRepositoryClassDefinition(com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest):	Caused an ERROR
    [junit] null
    [junit] java.lang.NullPointerException
    [junit] 	at com.liferay.portal.kernel.service.CompanyLocalServiceUtil.getCompanyByWebId(CompanyLocalServiceUtil.java:441)
    [junit] 	at com.liferay.portal.kernel.test.util.TestPropsValues.getCompanyId(TestPropsValues.java:78)
    [junit] 	at com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest.testGetRepositoryClassDefinition(RepositoryClassDefinitionCatalogImplTest.java:122)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    [junit] 
    [junit] 
    [junit] Testcase: testGetExternalRepositoryClassNames(com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest):	Caused an ERROR
    [junit] null
    [junit] java.lang.NullPointerException
    [junit] 	at com.liferay.portal.kernel.service.CompanyLocalServiceUtil.getCompanyByWebId(CompanyLocalServiceUtil.java:441)
    [junit] 	at com.liferay.portal.kernel.test.util.TestPropsValues.getCompanyId(TestPropsValues.java:78)
    [junit] 	at com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest.testGetExternalRepositoryClassNames(RepositoryClassDefinitionCatalogImplTest.java:100)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
    [junit] 
    [junit] 
    [junit] Test com.liferay.portal.repository.registry.RepositoryClassDefinitionCatalogImplTest FAILED
```

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by WYZ-Issac** [comment]
ci:test:sf

**Comment by WYZ-Issac** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:forward:force

**Pull request closed without merging**